clio-client/src/main/scala/org/broadinstitute/clio/client/util/IoUtil.scala

```diff
@@ -54,163 +52,94 @@ trait IoUtil {
       .fold(Seq.empty[Throwable]) { (acc, attempt) =>
         attempt.fold(acc :+ _, _ => acc)
       }
-      .flatMapConcat { errs =>
-        if (errs.isEmpty) {
-          Source.single(())
-        } else {
-          Source.failed(
-            new IOException(
-              s"""Failed to delete cloud objects:
-                 |${errs.map(_.getMessage).mkString("\n")}""".stripMargin
-            )
-          )
-        }
+      .flatMapConcat {
+        case Seq() => Source.single(())
+        case head +: tail =>
+          val exception =
+            new IOException("Failed to delete cloud objects", head)
+          tail.foreach(exception.addSuppressed)
+          Source.failed(exception)
       }
   }
 
-  /*
-   * FIXME for all below:
-   *
-   * Add a "credentials" argument (explicit or implicit) and
-   * use it to authenticate to GCS so we can convert the URIs
-   * into Paths and use the NIO API.
-   *
-   * We can't just parse as Paths from the start because the
-   * google-cloud-nio adapter will eagerly check storage authentication
-   * for each path, and fail.
-   */
-
-  def readGoogleObjectData(location: URI): String = {
-    gsUtil.cat(location.toString)
-  }
+  def readGoogleObjectData(location: URI): String
 
-  def copyGoogleObject(from: URI, to: URI): Unit = {
-    if (gsUtil.cp(from.toString, to.toString) != 0) {
-      throw new IOException(s"Failed to copy $from to $to in the cloud.")
-    }
-  }
+  def writeGoogleObjectData(data: String, location: URI): Unit
 
-  def deleteGoogleObject(path: URI): Unit = {
-    if (gsUtil.rm(path.toString) != 0) {
-      throw new IOException(s"Failed to delete $path in the cloud.")
-    }
-  }
+  def copyGoogleObject(from: URI, to: URI): Unit
 
-  def googleObjectExists(path: URI): Boolean = {
-    gsUtil.exists(path.toString) == 0
-  }
+  def deleteGoogleObject(path: URI): Unit
 
-  def getMd5HashOfGoogleObject(path: URI): Option[Symbol] = {
```

Comment:
Other projects depend on md5 hashes.

Reply:
There is an API to get it, so it's easy to implement if we need it. What is using it? I looked at backfill and workflow_tests and didn't see it used there. It's not used in clio. The backfill does override this method in `NioUtil` but it's not called, as far as I can tell.

Reply:
I thought the backfill uses it when backfilling wgs ubams. maybe we stopped doing that?

Reply:
There's a separate API that gets existence, size, and md5 all at once.

Either way I think it's OK to remove things here that the main clio code doesn't use, as long as the backfill code can extend `IoUtil` with new methods.

Reply:
The code in backfill uses `getGoogleObjectInfo()`, which is implemented using the storage API, not gsutil or the NIO API. That seems OK although there are some small overlaps with the code here and the code in backfill; we can decide what to do about backfill in a later checkin. Once this settles down I'll see about updating backfill.